app/models/device_token.rb

```diff
@@ -0,0 +1,16 @@
+class DeviceToken < ActiveRecord::Base
+  attr_accessible :token
+
+  def save
+    # Escape token: remove arrows and spaces
+    escaped_token = ""
+    token.each_char do |c|
+      escaped_token += c if c != " " && c != "<" && c != ">"
+    end
+    self.token = escaped_token
+
+    # Save if it does not already exist
+    return if DeviceToken.where("token = ?", self.token).count > 0
```

Comment:
Well, it is a bit complex with the phones. The token is sent from the phone every time you start the application (not resuming). 
I cannot rely on the phone for not sending me this token twice.